fv/infrastructure/infra_k8s.go

```diff
@@ -365,6 +378,17 @@ func (kds *K8sDatastoreInfra) EnsureReady() {
 }
 
 func (kds *K8sDatastoreInfra) Stop() {
+	// We don't actually stop the infra between tests because it's too expensive.
+	// Instead, mark all our resources for cleanup, but defer the cleanup until the
+	// start of the next test (this allows us to skip the cleanup if we happen to
+	// be the last test to run, which is a big win when manually running a single
+	// test for debugging.)
```

Comment:
Do we still need this, given the other speedups you've made today?  I'm a bit worried about the complexity, unless it's still a big win to do this as well.

Reply:
With the other fixes, my test run is down to about 25s; this shaves another 2-3s off, which feels worth it to me.  Another 2-3 similar optimisations and the FVs would be downright snappy :-)

Reply:
OK, fair enough!